cdk/src/open-data-platform/data-plane/data-import/write-demographic-data.handler.ts

```diff
@@ -83,255 +75,13 @@ function getTableRowFromRow(row: any): SqlParametersList {
-  numberOfRowsToWrite: number,
-): Promise<number> {
-  return new Promise(function(resolve, reject) {
-    let numberRowsParsed = 0;
-    const promises: Promise<BatchExecuteStatementResponse | null>[] = [];
-    const fileStream = S3.getObject(s3Params).createReadStream();
-
-    let pipeline = chain([
-      fileStream,
-      parser(),
-      pick({ filter: 'features' }),
-      streamArray(),
-      new Batch({ batchSize: BATCH_SIZE }),
-    ]);
-
-    try {
-      pipeline
-        .on('data', (rows: any[]) => {
-          // Stop reading stream if this would exceed number of rows to write.
-          if (numberRowsParsed + rows.length > numberOfRowsToWrite) {
-            pipeline.destroy();
-          }
-
-          let tableRows: SqlParametersList[] = [];
-          const shouldWriteRows = numberRowsParsed < numberOfRowsToWrite;
-
-          if (shouldWriteRows) {
-            tableRows = rows.map(getTableRowFromRow);
-          }
-
-          promises.push(executeBatchOfRows(rdsDataService, tableRows));
-          numberRowsParsed += rows.length;
-        })
-        .on('error', (error: Error) => {
-          reject(error);
-        })
-        // Gets called by pipeline.destroy()
-        .on('close', async (_: Error) => {
-          await handleEndOfFilestream(promises);
-          resolve(numberRowsParsed);
-        })
-        .on('end', async () => {
-          await handleEndOfFilestream(promises);
-          resolve(numberRowsParsed);
-        });
-    } catch (error) {
-      reject(error);
-    }
-  });
-}
-
-/**
- * Writes the table rows to the db and adds execution to list of promises.
- *
- * @param rdsDataService: RDS service to connect to the db.
- * @param tableRows: Rows to write to the db.
- */
-function executeBatchOfRows(
-  rdsDataService: RDSDataService,
-  tableRows: SqlParametersList[],
-): Promise<BatchExecuteStatementResponse | null> {
-  let promise: Promise<BatchExecuteStatementResponse | null> = Promise.resolve(null);
-  if (tableRows.length > 0) {
-    promise = insertBatch(rdsDataService, tableRows);
-    promise.catch((_) => {
-    }); // Suppress unhandled rejection.
-  }
-  return promise;
-}
-
-/**
- * Logs number and details of errors that occurred for all inserts.
- * @param promises of SQL executions.
- */
-async function handleEndOfFilestream(promises: Promise<BatchExecuteStatementResponse | null>[]) {
-  // Unlike Promise.all(), which rejects when a single promise rejects, Promise.allSettled
-  // allows all promises to finish regardless of status and aggregated the results.
-  const result = await Promise.allSettled(promises);
-  const errors = result.filter((p) => p.status == 'rejected') as PromiseRejectedResult[];
-  console.log(`Number errors during insert: ${errors.length}`);
-
-  // Log all the rejected promises to diagnose issues.
-  errors.forEach((error) => console.log(`Insert error: ${error}`));
-}
-
-
 /**
  * Parses S3 'alabama_acs_data.csv' file and writes rows
```

Comment:
I know you didn't add this but could you correct this filename?

Reply:
Done